tests/rfc003_erc20_htlc.rs

```diff
@@ -553,3 +580,135 @@ fn given_short_zero_secret_htlc_should_not_redeem() {
     );
     assert_eq!(client.token_balance_of(token_contract, bob), U256::from(0));
 }
+
+#[test]
+fn given_invalid_secret_htlc_should_not_redeem_and_print_log() {
+    let docker = Cli::default();
+    let secret_vec = vec![
+        0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+        0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+    ];
+
+    let (alice, bob, htlc_address, token_contract, token_amount, client, _handle, _container) =
+        erc20_harness(&docker, Erc20HarnessParams::default());
+
+    // Fund erc20 htlc
+    client.sign_and_send(|nonce, gas_price| UnsignedTransaction {
+        nonce,
+        gas_price,
+        gas_limit: U256::from(100_000),
+        to: Some(token_contract),
+        value: U256::from(0),
+        data: Some(
+            Erc20Htlc::transfer_erc20_tx_payload(
+                TokenQuantity(token_amount.into()),
+                Address(htlc_address.into()),
+            )
+            .into(),
+        ),
+    });
+
+    assert_eq!(
+        client.token_balance_of(token_contract, htlc_address),
+        U256::from(400)
+    );
+    assert_eq!(
+        client.token_balance_of(token_contract, alice),
+        U256::from(600)
+    );
+    assert_eq!(client.token_balance_of(token_contract, bob), U256::from(0));
+
+    // Send short secret to contract
+    let transaction_receipt = client.send_data(htlc_address, Some(Bytes(secret_vec)));
+
+    assert_eq!(
+        client.token_balance_of(token_contract, htlc_address),
+        U256::from(400)
+    );
+    assert_eq!(client.token_balance_of(token_contract, bob), U256::from(0));
+    assert_eq!(client.token_balance_of(token_contract, bob), U256::from(0));
+
+    // Check log was emitted
+    assert_that(&transaction_receipt.logs).has_length(1);
+    let topic: H256 = WRONGSECRET_LOG_MSG.parse().unwrap();
+    assert_that(&transaction_receipt.logs[0].topics).has_length(1);
+    assert_that(&transaction_receipt.logs[0].topics).contains(topic);
+    assert_that(&transaction_receipt.logs[0].data).is_equal_to(Bytes(vec![]));
+}
+
+#[test]
+fn invalid_secret_should_return_value() {
```

Comment:
This only differs from the previous test as it assertions that it returns a value, right?

I know that merging these two contradicts the "one assertion rule" suggested in a previous comment. However, I'd argue that we are testing how the contract reacts to an invalid secret here. I believe it would be better to test that in a single test (with several assertions of how it should behave)

Reply:
Good point. Merged and assert affter each test.